### PR TITLE
:bug: stop impersonating if switching apps

### PR DESCRIPTION
**Pull request by bjorngoa**

### CODE REVIEW

**Review by mariush2 — CHANGES_REQUESTED** [review]

File: src/organisms/TopBar/Account/Account.tsx [hunk 1]
```diff
@@ -96,6 +100,20 @@ export const Account: FC<AccountProps> = ({
...
+
+  useEffect(() => {
+    if (
+      activeImpersonationUser &&
+      activeImpersonationUser.appName != undefined &&
+      activeImpersonationUser.appName != APPLICATION_NAME
+    ) {
+      handleOnEndImpersonation();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (bjorngoa) [reply src/organisms/TopBar/Account/Account.tsx:115]: image

File: src/organisms/TopBar/Account/Account.tsx [hunk 2]
```diff
@@ -96,6 +100,20 @@ export const Account: FC<AccountProps> = ({
...
+  }, [activeImpersonationUser, APPLICATION_NAME]);
```
Comment (mariush2) [inline src/organisms/TopBar/Account/Account.tsx:116]:
```suggestion
  useEffect(() => {
    if (
      activeImpersonationUser &&
      activeImpersonationUser.appName != undefined &&
      activeImpersonationUser.appName != APPLICATION_NAME
    ) {
      endImpersonation();
    }
  }, [activeImpersonationUser, APPLICATION_NAME, endImpersonation]);
```

File: src/organisms/TopBar/Account/Account.tsx [hunk 3]
```diff
@@ -68,12 +70,14 @@ export const Account: FC<AccountProps> = ({
...
     import.meta.env.VITE_API_CLIENT_ID
   );
+  const APPLICATION_NAME = environment.getAppName(import.meta.env.VITE_NAME);
   const { account, roles, logout } = useAuth();
   const [isOpen, setIsOpen] = useState(false);
   const [openImpersonate, setOpenImpersonate] = useState(false);
   const buttonRef = useRef<HTMLButtonElement | null>(null);
   const { data: canImpersonate } = useCanImpersonate();
   const { data: activeImpersonationUser } = useActiveImpersonationUser();
+  const { mutateAsync: endImpersonation } = useStopImpersonation();
```
Comment (mariush2) [inline src/organisms/TopBar/Account/Account.tsx:80]:
```suggestion
  const { mutate: endImpersonation } = useStopImpersonation();
```

**Review by mariush2 — CHANGES_REQUESTED** [review]

File: src/organisms/TopBar/Account/Account.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mariush2) [reply src/organisms/TopBar/Account/Account.tsx:115]: Added a suggestion that should work

**Review by mariush2 — APPROVED** [review]

**Pull request merged**